### PR TITLE
Adds support for finding cudnn inside cuda path

### PR DESCRIPTION
**Pull request by fran6co**
## Description ##
Improves cuDNN check, most of the time it's in the same directory as CUDA.

### CODE REVIEW

**Comment by lupesko** [comment]
@cjolivier01 @anirudh2290 can you guys take a look please, help move this through?

**Comment by lebeg** [comment]
Well I'm slightly concerned about FirstClassCuda.cmake file in general. It is processed only in a few cases and in my oppinion should be removed completely: https://github.com/apache/incubator-mxnet/issues/12107

But for now if this change will help in some cases it's good.

File: cmake/FirstClassLangCuda.cmake [hunk 1]
```diff
@@ -32,12 +32,12 @@ function(detect_cuDNN)
...
 
   find_path(CUDNN_INCLUDE cudnn.h
-            PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT}
+            PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT} ${CUDA_TOOLKIT_ROOT_DIR}
             DOC "Path to cuDNN include directory." )
 
 
   find_library(CUDNN_LIBRARY NAMES libcudnn.so cudnn.lib # libcudnn_static.a
-                             PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT} ${CUDNN_INCLUDE}
+                             PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT} ${CUDNN_INCLUDE} ${CUDA_TOOLKIT_ROOT_DIR}
```
Comment (anirudh2290) [inline cmake/FirstClassLangCuda.cmake]: what scenario is this needed to make it work ? I don't have CUDNN_ROOT set but it still is able to find CUDNN_LIBRARY. is it when cuda is installed to a custom dir outside /usr or /usr/local

**Comment by stu1130** [comment]
@fran6co would you mind to address the feedback?
@mxnet-label-bot[pr-awaiting-response]

**Comment by vandanavk** [comment]
@sandeep-krishnamurthy  Please change label to pr-awaiting-response

**Comment by fran6co** [comment]
@vrakesh sorry but I don't have time to get back to this PR. Should I close it?

**Comment by sandeep-krishnamurthy** [comment]
@fran6co - Hope to see you contribute again!
Closing this PR for now due to staleness.

**Comment by szha** [comment]
@lebeg is this something you can pick up?

**Comment by lebeg** [comment]
@szha Yes, absolutely.

**Pull request closed without merging**